[14.0][FIX] shipment_advice: do not copy shipment advice id on moves

TDu opened the pull request:
In the case of partially receiving goods, the backorder moves created should not be included in the shipment advice being processed. But it probably makes sense for outgoing move as well.

mt-software-de reviewed (commented):
Thank you.

Should it also be `copy=False` on `stock.move.line`?

TDu commented:
> Thank you.
> 
> Should it also be `copy=False` on `stock.move.line`?

I do not think it is necessary, at least for what we are trying to fix here, which is handling backorder correctly.

mt-software-de commented:
I my opinion this should also be False for stock.move.line, it should be set by https://github.com/OCA/stock-logistics-transport/blob/487d71cce434bf0e817909f0208b64ec42de6c4d/shipment_advice/models/stock_move_line.py#L40 and not from a copy.

mt-software-de reviewed (approved):
LGTM

simahawk commented on shipment_advice/tests/test_shipment_advice.py:
```diff
@@ -246,3 +246,23 @@ def test_shipment_advice_draft(self):
         self.assertTrue("OUT" in self.shipment_advice_out.name)
         self.assertTrue("IN" in self.shipment_advice_in.name)
+
+    def test_shipment_advice_recieve_partial_move_backorder(self):
```
small typo
```suggestion
    def test_shipment_advice_receive_partial_move_backorder(self):
```

TDu replied:
Thanks, fixed.. and squashed.

simahawk reviewed (approved):
LG

mt-software-de commented:
@jbaudoux could you merge this one please?

mt-software-de commented:
> @jbaudoux could you merge this one please?

ping @OCA/logistics-maintainers ?

mt-software-de commented:
> What happens when you process with shopfloor and the move is split when you set destination ? Does it remains planned?

For shopfloor_delivery_shipment https://github.com/OCA/wms/blob/14.0/shopfloor_delivery_shipment/services/delivery_shipment.py we should be good.

For shopfloor_reception / shopfloor_reception_shipment_advice this could lead to an error, so i wouldn't remain planned. 

I think therefor we should extend `_prepare_move_split_vals` this would be used in https://github.com/OCA/wms/blob/7bca8fc3bf20d78a161010959c39edaca7c9bb8e/shopfloor_reception/services/reception.py#L1306 which is called by `_split`

Also there is `done_action` https://github.com/OCA/wms/blob/7bca8fc3bf20d78a161010959c39edaca7c9bb8e/shopfloor_reception/services/reception.py#L963 which validates the transfer.

mt-software-de commented:
> shopfloor_reception_shipment_advice needs to be adapted accordingly

I think we can overwrite `_prepare_move_split_vals` within this PR and set shipment advice id, if a specific context key is set. 
This context key can then be set in `shopfloor_reception_shipment_advice` https://github.com/OCA/wms/blob/7bca8fc3bf20d78a161010959c39edaca7c9bb8e/shopfloor_reception/services/reception.py#L963 and https://github.com/OCA/wms/blob/7bca8fc3bf20d78a161010959c39edaca7c9bb8e/shopfloor_reception/services/reception.py#L1282

@TDu when do you plan to do this?

TDu commented:
Rebased.
I have added a context key in `_prepare_move_split_vals` 
I am working on the change in `shopfloor_reception_shipment_advice` but struggling with  the tests there...

mt-software-de commented:
I just discussed this topic, again. And right now i don't know where this requirement is coming from anymore.
From my point of view i don't see any reason why the link shouldn't be copied within the reception scenario. 
We will schedule a meeting for this next week.

jbaudoux commented:
cc @rousseldenis (fyi)

rousseldenis commented on shipment_advice/models/stock_move.py:
```diff
@@ -23,3 +24,9 @@ def _prepare_merge_moves_distinct_fields(self):
+
+    def _prepare_move_split_vals(self, qty):
+        vals = super()._prepare_move_split_vals(qty)
+        if self.env.context.get("shipment_advice__propagate_on_split"):
```
@TDu @jbaudoux Why this key of context ?

Shouldn't it be possible to use the stock rule that has created the move and a property on it to do the same ?

mt-software-de replied:
Just as Information, we discussed this topic some weeks ago and there was no scenario where the id should not be copied. All of the cases in shopfloor as example would need the value. 
So i think we should close this one.

TDu replied:
It seems that this PR is not used on customer projects, I would say it can be closed.
Although I think the `copy=False` makes some sense.
